src/RecurringSwap.sol

```diff
@@ -105,8 +105,12 @@ contract RecurringSwap is QuarkScript {
             revert SwapWindowNotOpen(nextSwapTime, block.timestamp);
         }
 
-        // Update nextSwapTime
-        write(hashedConfig, bytes32(nextSwapTime + config.interval));
+        // Update the next swap time, ensuring that it is some time in the future
+        uint256 updatedNextSwapTime = nextSwapTime;
+        while (updatedNextSwapTime <= block.timestamp) {
+            updatedNextSwapTime += config.interval;
```

Comment:
The real question is if we should alter the logic to be `nextSwapTime = block.timestamp + config.interval`. The problem with that approach is it could lead to time drift, where a "Swap every Monday" slowly shifts to a "Swap every Tuesday" over time.

Reply:
Isn’t there a world where we check the window and mark that window as closed? E.g. if it’s Mondays starting Jan 1, then after the first Monday we say “next window=1” implying Jan 8. And if we miss a few we simply find the window we are now in and say “next window = current window + 1”?

Reply:
Discussed offline. We decided to add a `length` to the swap window. Basically, the swap is executable any time between the `swapWindowStartTime + swapWindowLength`. The `swapWindowStartTime` shifts in increments of `swapWindowInterval`.